tests/test_build_datapackage.py

```diff
@@ -56,7 +61,130 @@ def test_scalar_building():
+            )
+            process_mock.timeseries = pd.DataFrame()
+            return process_mock
+        elif process_name == "modex_tech_generator_gas":
+            process_mock = mock.Mock()
+            process_mock.scalars = pd.DataFrame(
+                {
+                    "region": {0: "BB", 1: "BB", 2: "BB"},
+                    "year": {0: 2016, 1: 2030, 2: 2050},
+                    "installed_capacity": {
+                        0: 330.2,
+                        1: 330.2014,
+                        2: 330.2014,
+                    },
+                    "emission_factor": {0: 0.2, 1: 0.2, 2: 0.2},
+                    "fuel_costs": {0: 25.9, 1: 25.9, 2: 49.36},
+                }
+            )
+            process_mock.timeseries = pd.DataFrame()
+            return process_mock
+        elif process_name == "modex_tech_wind_turbine_onshore":
+            process_mock = mock.Mock()
+            process_mock.scalars = pd.DataFrame(
+                {
+                    "region": {0: "BB", 1: "BB", 2: "BB"},
+                    "year": {0: 2016, 1: 2030, 2: 2050},
+                    "installed_capacity": {
+                        0: 5700.03,
+                        1: 5700.03375,
+                        2: 5700.03375,
+                    },
+                    "fixed_costs": {0: 23280.0, 1: 12600.0, 2: 11340.0},
+                    "lifetime": {0: 25.4, 1: 30.0, 2: 30.0},
+                    "wacc": {0: 0.07, 1: 0.07, 2: 0.07},
+                }
+            )
+            process_mock.timeseries = pd.DataFrame(
+                {
+                    "region": {0: "BB", 1: "BB", 2: "BB"},
+                    "timeindex_start": {
+                        0: "2016-01-01T00:00:00",
+                        1: "2030-01-01T00:00:00",
+                        2: "2050-01-01T00:00:00",
+                    },
+                    "timeindex_stop": {
+                        0: "2016-01-01T02:00:00",
+                        1: "2030-01-01T02:00:00",
+                        2: "2050-01-01T02:00:00",
+                    },
+                    "timeindex_resolution": {
+                        0: "P0DT01H00M00S",
+                        1: "P0DT01H00M00S",
+                        2: "P0DT01H00M00S",
+                    },
+                    "onshore": {
+                        0: [0.0516, 0.051, 0.0444],
+                        1: [0.0516, 0.051, 0.0444],
+                        2: [0.0516, 0.051, 0.0444],
+                    },
+                    "version": {0: "v8", 1: "v8", 2: "v8"},
+                }
+            )
+            return process_mock
+
+    mock_adapter.get_process.side_effect = mock_get_process
+    # Call the method with the mock adapter
+    result = DataPackage.build_datapackage(mock_adapter)
+    result.save_datapackage_to_csv("_files/build_datapackage_test")
+
+    check_if_csv_dirs_equal(
+        "_files/build_datapackage_goal", "_files/build_datapackage_test"
+    )
+
+
```

Comment:
what about checking the json?

Reply:
Yes that is still missing, thank you! Will add.